src/mbgl/map/transform_state.cpp

```diff
@@ -132,7 +132,9 @@ void TransformState::getProjMatrix(mat4& projMatrix, uint16_t nearZ, bool aligne
     matrix::rotate_z(projMatrix, projMatrix, getBearing() + getNorthOrientationAngle());
 
     const double dx = pixel_x() - size.width / 2.0f, dy = pixel_y() - size.height / 2.0f;
-    matrix::translate(projMatrix, projMatrix, dx, dy, 0);
+    if (!zero_centered) {
+        matrix::translate(projMatrix, projMatrix, dx, dy, 0);
```

Comment:
Do you have any example cases for this?

Reply:
I don't think you need to add a special condition for this in `getProjMatrix`. You should modify positions of your renderables in world space instead of negating the camera position. Floating point precision issues are also fixed by rendering relative to the camera.